sdk/keyvault/azure-keyvault-secrets/azure/keyvault/secrets/_models.py

```diff
@@ -69,10 +69,10 @@ def content_type(self):
 
     @property
     def id(self):
-        # type: () -> str
+        # type: () -> Optional[str]
```

Comment:
Since we're enforcing that the `attributes` and `vault_id` given to `__init__` are non-`None`, this property can assume it's non-`None` and we can remove any `if self._attributes` and `if self._vault_id` checks in properties